client/state/sites/guided-transfer/test/reducer.js

```diff
@@ -0,0 +1,147 @@
+	GUIDED_TRANSFER_STATUS_REQUEST,
+	GUIDED_TRANSFER_STATUS_REQUEST_FAILURE,
+	GUIDED_TRANSFER_STATUS_REQUEST_SUCCESS,
+	SERIALIZE,
+} from 'state/action-types';
+import reducer, {
+	status,
+	isFetching,
+} from '../reducer';
+
+describe( 'reducer', () => {
+	const testSiteId = 100658273;
+
+	before( () => {
+		sinon.stub( console, 'warn' );
+	} );
+
+	after( () => {
+		console.warn.restore();
+	} );
+
+	it( 'should include expected keys in return value', () => {
+		expect( reducer( undefined, {} ) ).to.have.keys( [
+			'status',
+			'isFetching'
+		] );
+	} );
+
+	describe( '#status()', () => {
+		it( 'should default to {}', () => {
+			const state = status( undefined, {} );
+
+			expect( state ).to.eql( {} );
+		} );
+
+		it( 'should store the status received', () => {
+			const guidedTransferStatus = {
+				upgrade_purchased: true,
+				host_details_entered: false,
+			};
+
+			const state = status( {}, {
+				type: GUIDED_TRANSFER_STATUS_RECEIVE,
+				siteId: testSiteId,
+				guidedTransferStatus,
+			} );
+
+			expect( state[ testSiteId ] ).to.eql( guidedTransferStatus );
+		} );
+
+		describe( 'persistence', () => {
+			it( 'persists state', () => {
+				const original = deepFreeze( {
+					[ testSiteId ]: {
+						upgrade_purchased: true,
+						host_details_entered: false,
+					}
+				} );
+				const state = status( original, { type: SERIALIZE } );
+				expect( state ).to.eql( original );
+			} );
+
+			it( 'loads valid persisted state', () => {
+				const original = deepFreeze( {
+					[ testSiteId ]: {
+						upgrade_purchased: true,
+						host_details_entered: false,
+					}
+				} );
+				const state = status( original, { type: DESERIALIZE } );
+				expect( state ).to.eql( original );
+			} );
+
+			it( 'loads default state when schema does not match', () => {
+				const original = deepFreeze( {
+					invalid_id: {
+						upgrade_purchased: true,
+						host_details_entered: false,
+					}
+				} );
+				const state = status( original, { type: DESERIALIZE } );
+				expect( state ).to.eql( {} );
+			} );
+		} );
+	} );
+
+	describe( '#isFetching()', () => {
+		it( 'should default to {}', () => {
+			const state = isFetching( undefined, {} );
+
+			expect( state ).to.eql( {} );
+		} );
+
+		it( 'should be true after a request begins', () => {
+			const state = isFetching( false, {
+				type: GUIDED_TRANSFER_STATUS_REQUEST,
+				siteId: testSiteId
+			} );
+			expect( state[ testSiteId ] ).to.eql( true );
+		} );
+
+		it( 'should be false when a request completes', () => {
+			const state = isFetching( true, {
+				type: GUIDED_TRANSFER_STATUS_REQUEST_SUCCESS,
+				siteId: testSiteId,
+			} );
+			expect( state[ testSiteId ] ).to.eql( false );
+		} );
+
+		it( 'should be false when a request fails', () => {
+			const state = isFetching( true, {
+				type: GUIDED_TRANSFER_STATUS_REQUEST_FAILURE,
+				siteId: testSiteId
+			} );
+			expect( state[ testSiteId ] ).to.eql( false );
+		} );
+
+		it( 'should never persist state', () => {
```

Comment:
Not sure how others feel about it and it certainly doesn't hurt to be thorough, but when using `createReducer`, I don't feel the need to test for the serialization behavior, as that should be guaranteed by the [tests for that utility](https://github.com/Automattic/wp-calypso/blob/1da2786823a1e596b12634e4cf6323a1233bd313/client/state/test/utils.js#L137-L147).